field/VerticalStaggerLoc.F90

```diff
@@ -80,4 +82,36 @@ elemental logical function are_not_equal(this, that)
       are_not_equal = .not. (this == that)
    end function are_not_equal
 
+   function get_dimension_name(this) result(dim_name)
+      character(:), allocatable :: dim_name
+      class(VerticalStaggerLoc), intent(in) :: this
+
+      select case (this%to_string())
+      case ("VERTICAL_STAGGER_NONE")
+         dim_name = ""
+      case ("VERTICAL_STAGGER_EDGE")
+         dim_name = "edge"
+      case ("VERTICAL_STAGGER_CENTER")
+         dim_name = "center"
+      case default
+         dim_name = "invalid"
+      end select
+   end function get_dimension_name
+
+   integer function get_num_levels(this, num_vgrid_levels) result(num_levels)
+      class(VerticalStaggerLoc), intent(in) :: this
+      integer, intent(in) :: num_vgrid_levels
+
+      select case (this%to_string())
+      case ("VERTICAL_STAGGER_NONE")
+         num_levels = 0
+      case ("VERTICAL_STAGGER_EDGE")
+         num_levels = num_vgrid_levels
+      case ("VERTICAL_STAGGER_CENTER")
+         num_levels = num_vgrid_levels - 1
+      case default
+         num_levels = -1
+      end select
+   end function get_num_levels
```

Comment:
1. Could you use this%id for the select case in both functions? You could use module private integer parameters for the id values so that the constructed instances and these functions are consistent.
2. Also, would it be worthwhile to make dim_name an attribute of VerticalStaggerLoc that is set at construction? It would add an argument to the constructor and attribute for the instance, but it would simplify the get_dimension_name.

Reply:
You've got it right.  I did not use `id` because wanted to avoid magic numbers.  And introducing named parameters gets weird here given the string names.    Would have done it anyway, but for the fact the need for this was last minute (see previous response).

Taking a note to fix this in next PR.

Reply:
Regarding (2).   I don't like the idea of "overspecifying" the value of an object.    Becomes more difficult to ensure consistency if someone gets creative later on.    I agree that it would simplify the new method.  OTOH, once I introduce the integer named parameter, this just becomes a SELECT CASE that can be easily verified by eye.

But the dim names should be params as well.  Adding that to my TODO list.

Reply:
fixed in next push.